addons/web/static/src/search/search_bar/search_bar.js

```diff
@@ -98,17 +99,39 @@ export class SearchBar extends Component {
             return;
         }
 
-        for (const searchItem of this.searchItems) {
+        // move properties under their related field
+        const searchItems = [...this.searchItems];
+        searchItems.sort((a, b) => (a.propertyItemId || a.id) - (b.propertyItemId || b.id));
```

Comment:
The thing is that we need to fetch the properties lazily, only if we unfold the properties field in the search bar.

That way, we won't do any extra HTTP request if we never search on properties.

The other thing is that the position in the `searchItems` is also important, beacsue all the current code base use id (which is not really an id, but an index...) and so we can't change the position in searchItems after fetching the properties (because it's done lazily)

So instead, we change the order when we generate the `items` list (the list used by the component), so have the proper order in the list

About sorting in `getSearchItems`, it's only used when the component is instanciate, but for our use use, we dynamically add searchItems based on user interaction, after the component is loaded

So in short the constraints here are that we
- absolutely need serachItem and not subItems (so we use the same code as normal field)
- we need to generate those serachItems lazily (so no extra HTTP request if we don't unfold properties)
- the id is in reality an index in the serachItems array (so we can't change it based on a user interraction)